ext/intl/bidi/bidi.c

```diff
@@ -63,48 +71,85 @@ static inline void php_bidi_throw_failure(php_intl_bidi_object *objval,
 }
 /* }}} */
 
+static inline bidi_object * bidi_create_bidi_object(zend_long maxRunCount, zend_long maxLength) {
+	UErrorCode error;
+	bidi_object * obj = ecalloc(1, sizeof(bidi_object));
+
+	if (PG(memory_limit) > 0) {
+		if (maxLength == 0) {
+			maxLength = PG(memory_limit) / 2;
+		} else if (maxLength > PG(memory_limit)) {
+			php_intl_bidi_throw_failure(NULL, U_ILLEGAL_ARGUMENT_ERROR,
+				"IntlBidi::__construct() given maxLength greater than memory_limit");
+			return NULL;
+		}
+	}
+
+	obj->childCount = 1;
+	intl_error_init(&(obj->error));
+
+	error = U_ZERO_ERROR;
+	obj->bidi = ubidi_openSized(maxLength, maxRunCount, &error);
+	if (U_FAILURE(error)) {
+		THROW_UFAILURE(NULL, "__construct", error);
+		return NULL;
+	}
+
+	return obj;
+}
+
+static inline void bidi_free_bidi_object(bidi_object * obj) {
+	if (obj != NULL) {
+		obj->childCount--;
+		if (obj->childCount == 0) {
+			if (obj->bidi) { ubidi_close(obj->bidi); }
+			if (obj->prologue) { efree(obj->prologue); }
+			if (obj->text)     { efree(obj->text); }
+			if (obj->epilogue) { efree(obj->epilogue); }
+			if (obj->embeddingLevels) { efree(obj->embeddingLevels); }
+
+			intl_error_reset(&(obj->error));
+
+			bidi_free_bidi_object(obj->parent);
+			obj->parent = NULL;
+
+			efree(obj);
+		}
+	}
+}
+
+static inline void php_intl_bidi_invokeConstruction(zval * instance, zend_long maxRunCount, zend_long maxLength) {
+	php_intl_bidi_object *objval = bidi_object_from_zend_object(Z_OBJ_P(instance));
+	objval->bidi = bidi_create_bidi_object(0, 0);
```

Comment:
You seem to be throwing maxRunCount and maxLength away. Is that intentional?

Reply:
No, but i fixed it in the lastest commit.